webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx

```diff
@@ -31,61 +39,59 @@ export default class CustomEnableDisableGuestAccountsSetting extends React.PureC
                 />
             );
         }
-        this.props.onChange(id, value, confirmNeeded, submit, warning);
-    };
+        onChange(id, value, confirmNeeded, submit, warning);
+    }, [onChange]);
 
-    public render() {
-        const label = (
-            <FormattedMessage
-                id='admin.guest_access.enableTitle'
-                defaultMessage='Enable Guest Access: '
+    const label = (
+        <FormattedMessage
+            id='admin.guest_access.enableTitle'
+            defaultMessage='Enable Guest Access: '
+        />
+    );
+
+    const helpText = (
+        <FormattedMarkdownMessage
+            id='admin.guest_access.enableDescription'
+            defaultMessage='When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests.'
+        />
+    );
+
+    return (
+        <>
+            <BooleanSetting
+                id={id}
+                value={value}
+                label={label}
+                helpText={helpText}
+                setByEnv={setByEnv}
+                onChange={handleChange}
+                disabled={disabled}
             />
-        );
-        const helpText = (
-            <FormattedMarkdownMessage
-                id='admin.guest_access.enableDescription'
-                defaultMessage='When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests.'
+            <ConfirmModal
+                show={showConfirm && (value === false)}
+                title={
+                    <FormattedMessage
+                        id='admin.guest_access.disableConfirmTitle'
+                        defaultMessage='Save and Disable Guest Access?'
+                    />
+                }
+                message={
+                    <FormattedMessage
+                        id='admin.guest_access.disableConfirmMessage'
+                        defaultMessage='Disabling guest access will revoke all current Guest Account sessions. Guests will no longer be able to login and new guests cannot be invited into Mattermost. Guest users will be marked as inactive in user lists. Enabling this feature will not reinstate previous guest accounts. Are you sure you wish to remove these users?'
+                    />
+                }
+                confirmButtonText={
+                    <FormattedMessage
+                        id='admin.guest_access.disableConfirmButton'
+                        defaultMessage='Save and Disable Guest Access'
+                    />
+                }
+                onConfirm={() => handleChange(id, false, true)}
```

Comment:
This generates a new function on every render. Can you move it to a `useCallback`?